packages/contract/src/contract-factory.ts

```diff
@@ -235,7 +235,11 @@ export default class ContractFactory {
       return { contract, transactionResult };
     };
 
-    return { waitForResult, contractId, transactionId: transactionResponse.id };
+    return {
+      contractId,
+      getTransactionId: () => Promise.resolve(transactionResponse.id),
```

Comment:
Did you consider following the `waitForResult` convention?

```ts
const { waitForResult, waitForTxId } = await factory.deploy();

const transactionId = await waitForTxId();
```

Reply:
I did not but this is better 👍🏻

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2929/commits/b6814ce7bf26b16ba44712356f4027c7655a78c8